server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -234,14 +238,26 @@ static SecureSettings loadSecureSettings(Environment initialEnv) throws Bootstra
             throw new BootstrapException(e);
         }
 
-        try {
+        char[] passChars;
+        try (BufferedReader b = new BufferedReader(new InputStreamReader(System.in, StandardCharsets.UTF_8))) {
```

Comment:
@jkakavas Ioannis — what do you think? It seems as if right now our tools accept anything a user types in for a password, up to a newline. Should we define some limits on length and characters?

Reply:
I'll pose no objections to adding a max length for keystore passwords, if this is significantly simplifies our code, assuming that we aim for a futureproof limit of, let's say, 128 chars. 

It's not unheard of to use maximum password length validation. This is mostly relevant for authentication as they're in place to prevent resource depletion DoS attacks. `PBKDF2WithHmacSHA512` sets no limits for the input password. I _randomly_ picked 128 as it matches OWASP recommendation for authentication systems, it allows the use of long passphrases and I don't see why we should limit this more.